### DISCUSSION TITLE
[Request] Create match Discord/Telegram

### DISCUSSION THREAD

**Discussion by furzich2015** [PhlexPlexico/G5API#76] [Ideas]
Hey. I have an idea for a get5 system.
Why not create a match via discord / telegram?
The principle of how you can create a match:

1. A separate chat on the discord server, which will be created after the command! Start5on5
        1.1 NameTeam A: "A"
        1.2 NameTeam B: "B"
	1.3 1on1,2on2,5on5: "1on1"
	1.4 Bo1/Bo3/Bo5: "bo1"
	1.5 SteamId Team A: "******************************, *******************************,*****************************"
	1.6 SteamId Team B: "******************************, *******************************,*****************************"
	1.7 Pick-ban maps: "yes"
	1.8 Enable/Disable knife round: "Enable" 
	1.9 Record Demos: "yes"

Similarly, for the bot in Telegram.
This is a great misl in my head, but I don’t know how difficult it is to implement it?
And you have such an idea?

**Comment by PhlexPlexico** [comment]
https://github.com/PhlexPlexico/G5-Bot/tree/api 

Since this uses an API, all you need is an API key and you can do everything the front end can do. It will create a match with vetoes and allow you to choose a server. The server IDs are stored in the settings as well, and it will also grab whatever public servers are available :)

I haven't tested it, but it was something I was playing around with for the API to ensure everything could work with an API key.

**Comment by PhlexPlexico** [comment]
Also, gonna move this over to https://github.com/PhlexPlexico/G5API since it's more to do with the backend then the front end

**Comment by PhlexPlexico** [comment]
Also if you're looking for defined teams, I think the steps would be like this:

1. Create two teams with given names,
1.1 Provide the team auths
2. Create a match
2.2 Max players, max players to ready, best of series choice, veto in game, knife rounds etc go here.

And you would have to select the server ID as well. But it's totally possible, and is supported! All it is, is few POST calls away :)
